### PR TITLE
Fixes UI Bug Issue in Symptom Editor

### LINKED ISSUE #8998: Fix UI bugs with Symptoms editor and move to ModelCrudEditor component
Image

- Fix the overlap in date input
- Fix the dropdown icon.
- Port over the `<EncounterSymptomsBuilder>` component to use `<ModelCrudEditor>` 
Example: https://github.com/ohcnetwork/care_fe/blob/a0e6e621260a3501e3828c3ad9f14ae7efd3d94f/src/components/Facility/SpokeFacilityEditor.tsx#L132

### PR DESCRIPTION
**Pull request by JavidSumra**
## Proposed Changes

 - Fixes #8998
 - Align the dropdown Icon in the search for symptoms input field.
 - Fixed Placeholder in dateinput
 - Moved to ModelCrudEditor in EncounterSymptomsBuilder.

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Enhanced accessibility of the date input field with a tooltip on hover.
	- Improved layout and presentation of the Autocomplete Multiselect dropdown.
	- Updated styling for the DateFormField component.
	- Minor layout adjustments in the ModelCrudEditor component.

- **Bug Fixes**
	- Clarified default behavior for max and min date props in the DateFormField.

### CODE REVIEW

File: src/components/Symptoms/types.ts [hunk 1]
```diff
@@ -43,10 +43,20 @@ type ClinicalImpressionStatus =
   | "entered-in-error";
 
 export interface EncounterSymptom extends BaseModel {
-  symptom: (typeof SYMPTOM_CHOICES)[number]["id"];
+  symptom: (typeof SYMPTOM_CHOICES)[number]["id"] | "";
```
[the remark this answers is not in the document]
Comment (JavidSumra) [reply src/components/Symptoms/types.ts]: Looking into it

**Comment by JavidSumra** [comment]
Hey @shivankacker, it seems that when we update fields like `onsetDate` or `symptom`, the component gets re-rendered. This is expected behavior, so I believe the test suit should be updated to account for this. Would it be okay to adjust the test suit accordingly?

**Comment by shivankacker** [comment]
I don't think the tests are failing because of that. Go through test screenshots and see if you can find what is causing the failure

**Comment by nihal467** [comment]
@JavidSumra Can you please share the current status of the PR? Otherwise, the PR will be closed in the next 48 hours, and you will be unassigned from the issue due to inactivity.

**Comment by JavidSumra** [comment]
Hey @nihal467, I've completed the requested changes, but the test suite is failing due to some issues. I'm discussing this with Shivank to resolve it, so I'll need some additional time to fix it.

**Comment by JavidSumra** [comment]
Hey @nihal467, the `PatientLogUpdate` test case was failing due to some issues. I added a `cy.wait` before the date selection, and now it's passing. I also tried using the `cy.intercept` method, but it didn't work. Could you please guide me on the next steps?

**Comment by nihal467** [comment]
https://www.loom.com/share/b0d219030c524e4c8ca6a35660191ff4?sid=44912730-a263-4af0-a768-0af799822c5e

@JavidSumra  when we initially open the log update page, the component is auto closed without interaction

**Comment by JavidSumra** [comment]
@nihal467 need your review.

**Review by Jacobjeevan — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**